cuML/src/glm/qn/glm_base.h

```diff
@@ -169,7 +164,10 @@ template <typename T, class GLMObjective> struct GLMWithData : GLMDims {
     Mat G(gradFlat.data, C, dims);
```

Comment:
@jirikraus `SimpleMat` and `SimpleVec` are simple views on provided gpu memories. The only point in `glm/qn` folder where a mem alloc could happen is in the entry point in the API function in `qn.h` where the complete workspace is created/requested.

The conversion between Vec and Mat is just a wrapping for semantic reasons, because conceptually the non-linear solver optimizes functions on R^D, while the GLM loss functions are modelled to have parameter matrices (for the case of multi-task/multi-class problems).